nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -1640,6 +1640,76 @@ public void overrideWithRawType() {
         .doTest();
   }
 
+  @Test
+  public void testForNullableUpperBoundsInLibModel() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "import java.util.function.Function;",
+            "class Test {",
+            "  static String foo(@Nullable String a) {",
+            "    if(a!=null){ ",
+            "     return a.replace(\"a\",\"\");",
+            "    }else{",
+            "     return \"\";",
+            "    }",
+            "  }",
+            "  static void testPositiveMethodRef() {",
+            "   Function<String,@Nullable String> removeA = Test::foo;",
```

Comment:
@akulk022 something is wrong here.  The type of `Test::foo` is `Function<@Nullable String,String>`, but the type of `removeA` is `Function<String,@Nullable String>`.  So we should be getting an error at the assignment, in addition to an error at the `removeA.apply(null)` call.  (In fact I think it should show up as an "override" error when checking the `Test::foo` expression.  Can you see what is going on?

Reply:
Thinking more my comment is wrong.  The LHS type is `Function<String,@Nullable String>` which means a `@NonNull` parameter and a `@Nullable` return.  `foo` takes a `@Nullable` parameter and returns `@NonNull`.  This is valid function subtyping, so we're fine here.  My mistake.